discovery.md

```diff
@@ -432,78 +367,105 @@ requirements for the query API.
-  TODO: Needs resolution with CloudSubscriptions API
-- Constraints:
-  - OPTIONAL
-- Examples:
+### REST Paths
 
-###### authscope
+Each path in the REST API represents either a list (or search) over
+the Discovery Endpoint, or the retrieval of an individual entity. All of
+these operations MUST be supported by compliant discovery implementations.
 
-- Type: `String`
-- Description: Authorization scope needed for creating subscriptions.
-  The actual meaning of this field is determined on a per-producer basis.
-- Constraints:
-  - OPTIONAL
-- Example:
-  - "storage.read"
+Note: the relative paths specified below are NOT REQUIRED to be at the root
+of the `fpath` (per RFC1738). However, they are REQUIRED to match the end
+of it. For example, the follow are valid URLs/paths:
 
-### REST Paths
+```
+https://example.com/services
+https://example.com/myAggregator/services
+```
 
-Each path in the REST API represents either represents a list (or search) over
-and entity class, or the retrieval of an individual entity by name. Each of
-these operations MUST be supported by compliant discovery implementations.
+Note: for each query if the client is not authorized to see any particular
+entity then that entity MUST be excluded from the response. In cases where
+response is a single entity, then the response MUST result in an error
+as if the entity did not exist (e.g. for HTTP the response would be 
+`404 Not Found`). In cases where the response is an array, then the response
+MUST return a successful status with an array, even if that array is empty.
+
+#### Services
+
+#### `/services`
+
+This MUST return an array of zero or more Services. The list MUST contain
+all Services available via this Discovery Enpoint.
+
+When encoded in JSON, the response format MUST adhere to the following:
 
-#### `/services?matching=[search term]`
+```
+[
+  {
+    "url": "SERVICE-URL",
+    "name": "{name}",
+    ... remainder of Service attributes ...
+  }
+  ...
+]
+```
+
+##### `/services/{name}`
+
+If this refers to a valid Service, then this MUST return that single
+Service entity.
+
+When encoded in JSON, the response format MUST adhere to the following:
+
+```
+{
+  "url": "SERVICE-URL",
+  "name": "{name}",
+  ... remainder of Service attributes ...
+}
+```
 
-Returns a list of all services in the discovery system, optionally filtering on
-a provided search term (`matching`).
+##### `/services?matching=[search term]`
 
-##### matching
+Same as `/services` but the array MUST be limited to just those Services
+whose `name` attribute contains the `search term` value (case insensitive).
+
+###### matching
 
 * Type: `string`
 * Description: Search term that provides case insensitive match against
-  `service` names. The parameter can match any portion of the service name.
+  the Service's `name` attribute. The parameter can match any portion
+  of the service's `description` value.
```

Comment:
On the previous lines above you said 

> whose `name` attributes contains
<br>
but here you were saying

> The parameter can match any portion of the service's `description` value.

Was the intend that the result should return any match to `name` or `description`?

Reply:
Fixed - good catch. I switch from `description` to `name` late yesterday.